### PR TITLE
Installation checks if either configs ("enableProjectTypeInWorkspace", "enableWorkspaceFilesystem") are 'false' and makes them 'true' for valid installation

### LINKED ISSUE #576: Installation error - databricks.sdk.errors.mapping.NotFound: Files in Repos must be enabled for AUTO ExportFormat
Here is the stack trace -

```
Enter a number between 0 and 2: 2
Comma-separated list of workspace group names to migrate. If not specified, we'll use all account-level groups with matching names to workspace-level groups. (default: <ALL>): 
Backup prefix (default: db-temp-): 
Log level (default: INFO): 
Number of threads (default: 8): INFO [__main__] Creating configuration file: /Users/XXXXX.XXXXX@domain.com/.ucx/config.yml
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/m3210483/source/ucx/src/databricks/labs/ucx/install.py", line 837, in <module>
    installer.run()
  File "/home/m3210483/source/ucx/src/databricks/labs/ucx/install.py", line 107, in run
    self._configure()
  File "/home/m3210483/source/ucx/src/databricks/labs/ucx/install.py", line 367, in _configure
    self._write_config()
  File "/home/m3210483/source/ucx/src/databricks/labs/ucx/install.py", line 383, in _write_config
    self._ws.workspace.upload(self.config_file, config_bytes, format=ImportFormat.AUTO)
  File "/tmp/tmp.RALkKV481G/lib/python3.10/site-packages/databricks/sdk/mixins/workspace.py", line 90, in upload
    raise e
  File "/tmp/tmp.RALkKV481G/lib/python3.10/site-packages/databricks/sdk/mixins/workspace.py", line 84, in upload
    return self._api.do('POST', '/api/2.0/workspace/import', files={'content': content}, data=data)
  File "/tmp/tmp.RALkKV481G/lib/python3.10/site-packages/databricks/sdk/core.py", line 1070, in do
    return retryable(self._perform)(method,
  File "/tmp/tmp.RALkKV481G/lib/python3.10/site-packages/databricks/sdk/retries.py", line 47, in wrapper
    raise err
  File "/tmp/tmp.RALkKV481G/lib/python3.10/site-packages/databricks/sdk/retries.py", line 29, in wrapper
    return func(*args, **kwargs)
  File "/tmp/tmp.RALkKV481G/lib/python3.10/site-packages/databricks/sdk/core.py", line 1162, in _perform
    raise self._make_nicer_error(response=response, **payload) from None
databricks.sdk.errors.mapping.NotFound: Files in Repos must be enabled for AUTO ExportFormat
```

### PR DESCRIPTION
**Pull request by pritishpai**
## Changes
Installation fails if either of the configs ("enableProjectTypeInWorkspace", "enableWorkspaceFilesystem") are 'false'. Checks the two configs via API and sets them to "true" if they are "false". 

### Linked issues

Resolves #576

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [x] manually tested
- [x] added unit tests
- [x] added integration tests
- [ ] verified on staging environment (screenshot attached)

Using SDK for getting and setting conf values blocked by https://github.com/databricks/databricks-sdk-py/issues/500

### CODE REVIEW

File: src/databricks/labs/ucx/install.py [hunk 1]
```diff
@@ -1037,6 +1040,34 @@ def validate_and_run(self, step: str):
...
+        project_type = client.do("GET", "/api/2.0/workspace-conf", {"keys": "enableProjectTypeInWorkspace"})
+        workspace_file_system = client.do("GET", "/api/2.0/workspace-conf", {"keys": "enableWorkspaceFilesystem"})
+
+        logger.debug("Checking Files In Repos configuration")
+
+        if project_type["enableProjectTypeInWorkspace"] == "false":
+            logger.debug("enableProjectTypeInWorkspace is False, enabling the config")
+            client.do(
+                "PATCH",
+                "/api/2.0/workspace-conf",
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:1057]: Why ws.workspace_conf.set_status() doesn't work? Did you try that?

File: src/databricks/labs/ucx/install.py [hunk 2]
```diff
@@ -1037,6 +1040,34 @@ def validate_and_run(self, step: str):
         if not self.validate_step(step):
             self.run_workflow(step)
 
+    @staticmethod
+    def enable_files_in_repos():
+        client = ApiClient()
+
+        # check if "enableProjectTypeInWorkspace" and "enableWorkspaceFilesystem" are set to false
+        project_type = client.do("GET", "/api/2.0/workspace-conf", {"keys": "enableProjectTypeInWorkspace"})
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:1048]: Get two keys at once, you can get that by separating them with a comma

**Review by nfx — CHANGES_REQUESTED** [review]
Use sdk, not api

File: src/databricks/labs/ucx/install.py (hunk 1 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/install.py:1057]: Blocked by https://github.com/databricks/databricks-sdk-py/issues/500

**Comment by nfx** [comment]
sdk is fixed - https://github.com/databricks/databricks-sdk-py/pull/525

**Comment by pritishpai** [comment]
The sdk changes are not released yet, so make changes to the toml file and pulled in latest commit. Can only merge after the next release. There are other changes in sdk that I had to make to update the codebase(e.g. [https://github.com/databricks/databricks-sdk-py/pull/522]). Should I added them as a part of this PR too @nfx ?

File: src/databricks/labs/ucx/install.py [hunk 3]
```diff
@@ -339,6 +339,7 @@ def folder(self):
 
     def run(self):
         logger.info(f"Installing UCX v{PRODUCT_INFO.version()}")
+        self._enable_files_in_repos()
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:342]: Can we do it on the installer side (before we save() config)? Or even better - in blueprint, when we receive error that feature is disabled?.. do you remember the error code for that?

Comment (pritishpai) [reply src/databricks/labs/ucx/install.py:371]: This was the error:
 databricks.sdk.errors.mapping.NotFound: Files in Repos must be enabled for AUTO ExportFormat

File: src/databricks/labs/ucx/install.py [hunk 4]
```diff
@@ -368,6 +368,7 @@ def folder(self):
...
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:371]: Can you move this to WorkspaceInstaller, before you read the current configuration file

File: src/databricks/labs/ucx/install.py [hunk 5]
```diff
@@ -962,6 +963,21 @@ def validate_and_run(self, step: str):
         if not self.validate_step(step):
             self.run_workflow(step)
 
+    def _enable_files_in_repos(self):
+        # check if "enableProjectTypeInWorkspace" and "enableWorkspaceFilesystem" are set to false
+        project_type = self._ws.workspace_conf.get_status("enableProjectTypeInWorkspace")
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:968]: Supply comma-separated list of settings names to make just one api call.

Why do we need project type (git repos), by the way?

File: src/databricks/labs/ucx/install.py [hunk 6]
```diff
@@ -962,6 +963,21 @@ def validate_and_run(self, step: str):
...
+
+        logger.debug("Checking Files In Repos configuration")
+
+        if project_type["enableProjectTypeInWorkspace"] == "false":
+            logger.debug("enableProjectTypeInWorkspace is False, enabling the config")
+            self._ws.workspace_conf.set_status("enableProjectTypeInWorkspace")
+
+        if workspace_file_system["enableWorkspaceFilesystem"] == "false":
+            logger.debug("enableWorkspaceFilesystem is False, enabling the config")
+            self._ws.workspace_conf.set_status("enableWorkspaceFilesystem")
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:979]: Method expects you to have dictionary https://databricks-sdk-py.readthedocs.io/en/latest/workspace/settings/workspace_conf.html#databricks.sdk.service.settings.WorkspaceConfAPI.set_status

File: tests/integration/test_installation.py [hunk 7]
```diff
@@ -310,3 +310,12 @@ def test_uninstallation(ws, sql_backend, new_installation):
         ws.jobs.get(job_id=assessment_job_id)
     with pytest.raises(NotFound):
         sql_backend.execute(f"show tables from hive_metastore.{install.config.inventory_database}")
+
+
+def test_files_in_repos_enablement(new_installation):
```
Comment (nfx) [inline tests/integration/test_installation.py:315]: We don't need this integration test

**Review by nfx — CHANGES_REQUESTED** [review]
Could you implement it in the blueprint library instead? Eg manually create a new dummy workspace (for isolation), disable settings, then in Installation.load (or deeper), catch the "feature disabled" exception, call this code to set the status, and retry there.

This would be the most impactful implementation, that is the easiest to test in isolation

**Comment by pritishpai** [comment]
Work now done in the blueprint repo:
https://github.com/databrickslabs/blueprint/pull/42

File: src/databricks/labs/ucx/install.py (hunk 5 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/install.py:968]: During initial testing both value were needed. I tested against a new workspace and just flipping enableWorkspaceFilesystem suffices.

File: src/databricks/labs/ucx/install.py (hunk 3 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/install.py:371]: Moved to blueprint

**Comment by pritishpai** [comment]
https://github.com/databrickslabs/blueprint/pull/42

**Pull request closed without merging**